### ISSUE TITLE
Splicing a macro inside an inline method does not work

### ISSUE THREAD

**Issue by martin-ockajak** [VirtusLab/scala-cli#2730]
**Version(s)**

Scala CLI - `1.1.2`
Scala Compiler - `3.3.1`

**Describe the bug**

Splicing a simple macro inside an inline method results in the following error suggesting the macro implementation is not a static method:
```scala
[error] ./macro-error.sc:7:6
[error] Malformed macro.
[error]
[error] Expected the splice ${...} to contain a single call to a static method.
[error]   ${ testMacro }
[error]      ^^^^^^^^^
Compilation failed
```

However the `testMacro` method should be static since it is defined like that. The same problem manifests if both methods are explicitly wrapped in an object.

**To Reproduce**

Run this script:
```scala
#!/usr/bin/env -S scala-cli -q
//> using scala 3.3.1

import scala.quoted.{Expr, Quotes, Type}

inline def test: Unit =
  ${ testMacro }

def testMacro(using quotes: Quotes): Expr[Unit] =
  '{ () }
```

**Expected behaviour**

The macro should compile without errors, just as the equivalent SBT project does.

**Note**

Thanks for making a fantastic tool 😁.

**Comment by bishabosha** [comment]
The thing is that with `.sc` files, the entire code is always wrapped in a class (unless with `//> using objectWrapper` directive), which means ultimately your macro methods will not be "java static". Unless you propose that top level classes/objects should be lifted out of the script and only expressions be in the wrapper

the other solution is to not use a script file, but a `.scala` file with a main method.


Ultimately, **you are not allowed to call a macro defined in the same file**, so this entire use case "single file with macro" is invalid for scripting (maybe `//> using file macros.scala` could work?)

**Comment by martin-ockajak** [comment]
@bishabosha I suspected that (wrapping the code in a class) might be the case, thanks for clarifying. The actual macro call is superfluous in this example (removed now), the compilation error happens without it too, for the reason you mentioned.

I suppose it boils down to a question of whether the class instance wrapping is necessary/desirable and whether it might perhaps lead to some other issues. In other words, is it a feature or a bug ?

**Comment by Gedochao** [comment]
> I suppose it boils down to a question of whether the class instance wrapping is necessary/desirable and whether it might perhaps lead to some other issues. In other words, is it a feature or a bug?

It is neither. 

Both the class wrapper and the object wrapper have their pros and cons, with neither being a flawless choice.
In Scala 3 we default to the class wrapper, as its problems aren't much of an issue for most use cases.
You can always use `//> using objectWrapper` when you deem it necessary.

There's more to read on this topic in [the relevant doc on script wrappers](https://scala-cli.virtuslab.org/docs/guides/scripting/scripts#script-wrappers).

**Comment by Gedochao** [comment]
TL;DR the behaviour is expected, I'm closing the issue.
Can refer to https://github.com/VirtusLab/scala-cli/issues/2247 on the known issues tied to script wrappers.
